### ISSUE TITLE
Mutating Values in onSubmit are reflecting in Form

### ISSUE THREAD

**Issue by santosh898** [jaredpalmer/formik#2346]
### Current Behavior

``` 
const onFormSubmit = async (values) => {
 delete values.sections;
};
```
This deletes the sections from values object in the original form.

### Expected behavior

Mutation to the values in `onSubmit` method shouln't affect the form.

### Reproducible example

https://codesandbox.io/s/formik-codesandbox-template-p03t9

### Suggested solution(s)

Make `values` argument to the `onSubmit` **immutable** or a **clone**

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik          | 2.1.4
| React            | 16.13.0

**Comment by mhelmer** [comment]
I don't think this is unexpected at all. If you decide to create a function that mutates its input, there is no reason to be surprised that it affects the outside world. This is the usual behavior. You can just clone the object yourself and make your changes to the clone.
